tests/tasks/setup.yml

```
@@ -14,3 +14,20 @@
   package:
     name: "{{ __ssh_test_packages }}"
     state: present
+
+- name: Define common variables
+  ansible.builtin.set_fact:
+    main_ssh_config: /etc/ssh/ssh_config
+    main_ssh_config_name: ssh_config
+    main_ssh_config_path: /etc/ssh/
+
+- name: Define specific variables
+  ansible.builtin.set_fact:
+    main_ssh_config: /etc/ssh/ssh_config.d/00-ansible.conf
+    main_ssh_config_name: 00-ansible.conf
+    main_ssh_config_path: /etc/ssh/ssh_config.d/
+  when:
+    - (ansible_facts['os_family'] == 'RedHat'
+        and ansible_facts['distribution_major_version']|int > 8) or
+      (ansible_facts['distribution'] == 'Ubuntu'
+        and ansible_facts['distribution_major_version']|int >= 22)
```

The client include was introduced much earlier than the server include so the RHEL8 already has the drop-in directory:

https://github.com/linux-system-roles/ssh/blob/main/vars/RedHat_8.yml

For Ubuntu, we don't have versions difference in the vars file so the version check should not really be there and something like the following should do:

https://github.com/linux-system-roles/ssh/blob/main/vars/Ubuntu.yml

```suggestion
    - (ansible_facts['os_family'] == 'RedHat'
        and ansible_facts['distribution_major_version']|int >= 8) or
     ansible_facts['distribution'] == 'Ubuntu'
```
(the new version of debian was released recently which might get this too so we might need to add that difference there in the end too, but it is for different PR).